docs/openstack/edpm_adoption.md

```diff
@@ -302,6 +267,202 @@ done
     ```
 
 * Wait for the dataplane node set to reach the Ready status:
+
     ```
     oc wait --for condition=Ready osdpns/openstack --timeout=30m
     ```
+
+## Nova compute services fast-forward upgrade from Wallaby to Antelope
+
+Nova services rolling upgrade cannot be done during adoption,
+there is in a lock-step with Nova control plane services, because those
+are managed independently by EDPM ansible, and Kubernetes operators.
+Nova service operator and OpenStack Dataplane operator ensure upgrading
+is done independently of each other, by configuring
+`[upgrade_levels]compute=auto` for Nova services. Nova control plane
+services apply the change right after CR is patched. Nova compute EDPM
+services will catch up the same config change with ansible deployment
+later on.
+
+> **NOTE**: Additional orchestration happening around the FFU workarounds
+> configuration for Nova compute EDPM service is a subject of future changes.
+
+* Configure pre-FFU workarounds for Nova compute EDPM services to update its version records:
+
+    ```yaml
+    oc apply -f - <<EOF
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: nova-compute-workarounds
+      namespace: openstack
+    data:
+      19-nova-compute-cell1-workarounds.conf: |
+        [workarounds]
+        disable_compute_service_check_for_ffu=true
+    EOF
+
+    oc apply -f - <<EOF
+    apiVersion: dataplane.openstack.org/v1beta1
+    kind: OpenStackDataPlaneService
+    metadata:
+      name: nova-compute-workarounds
+      namespace: openstack
+    spec:
+      label: nova.compute.workarounds
+      configMaps:
+        - nova-compute-workarounds
+      playbook: osp.edpm.nova
+    ---
+    apiVersion: dataplane.openstack.org/v1beta1
+    kind: OpenStackDataPlaneDeployment
+    metadata:
+      name: openstack-nova-compute-workarounds
+      namespace: openstack
+    spec:
+      nodeSets:
+        - openstack
+      servicesOverride:
+        - nova-compute-workarounds
+    EOF
+    ```
+
+* Wait for cell1 Nova compute EDPM services version updated (it may take some time):
+
+    ```bash
+    oc exec -it mariadb-openstack-cell1 -- mysql --user=root --password=${PODIFIED_DB_ROOT_PASSWORD} \
+      -e "select a.version from nova_cell1.services a join nova_cell1.services b where a.version!=b.version and a.binary='nova-compute';"
+    ```
```

Comment:
At this point you can observe the the compute was able to report status to the new control plane so the service is now UP:
```
[gibi@osp-dev-01 ~]$ openstack compute service list
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
| ID                                   | Binary         | Host                   | Zone     | Status  | State | Updated At                 |
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
| 0954e21c-9022-4718-a570-a7d3eb0fd79f | nova-conductor | nova-cell0-conductor-0 | internal | enabled | up    | 2023-11-09T09:51:43.000000 |
| c3ad2def-18ed-49e5-8af4-a7c1a0840171 | nova-scheduler | nova-scheduler-0       | internal | enabled | up    | 2023-11-09T09:51:39.000000 |
| a7a20d50-b85d-4321-a576-5a12fea9bc8f | nova-compute   | standalone.localdomain | nova     | enabled | up    | 2023-11-09T09:51:41.000000 |
| 9eb053f9-a404-4b02-92a8-d2a5fe339849 | nova-conductor | nova-cell1-conductor-0 | internal | enabled | up    | 2023-11-09T09:51:45.000000 |
+--------------------------------------+----------------+------------------------+----------+---------+-------+----------------------------+
[gibi@osp-dev-01 ~]$ openstack hypervisor list
+--------------------------------------+------------------------+-----------------+-----------------+-------+
| ID                                   | Hypervisor Hostname    | Hypervisor Type | Host IP         | State |
+--------------------------------------+------------------------+-----------------+-----------------+-------+
| d3d2be51-a0b9-4538-a298-62280a52fece | standalone.localdomain | QEMU            | 192.168.122.100 | up    |
+--------------------------------------+------------------------+-----------------+-----------------+-------+

```

Reply:
I don't get this, sorry. What is expected to change along these lines?